pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -140,9 +140,6 @@ func getFloatingIPByPortID(client *gophercloud.ServiceClient, portID string) (*f
 			return false, err
 		}
 		floatingIPList = append(floatingIPList, f...)
-		if len(floatingIPList) > 1 {
```

Comment:
This means the loop might keep going, only to error out later (which is fine and I agree it's an error case not worth optimising for).

Perhaps the final return line (L146) could become:
```go
  return len(floatingIPList) <= 1, nil
```
And this would achieve the same early-exit-on-multiple-results but with far less visual overhead.

---

Since this loop is such a common pattern in the openstack provider, an alternative that I've been meaning to do for a while would be to replace the whole lot with some sort of generic code that takes an extraction function (like `floatingips.ExtractFloatingIPs`), does this generic "build a list, and abort if you get more than one result" logic, and just returns a single (interface{}, error).  Again, I wish golang had parameterised types :(